src/transformers/models/swin/modeling_swin.py

```diff
@@ -42,6 +44,16 @@
 ]
 
 
+@dataclass
+class SwinModelOutputWithPooling(BaseModelOutputWithPooling):
+    hidden_states_original_spatial_dimensions: Tuple[Tuple[int, int]] = None
+
+
+@dataclass
+class SwinBaseModelOutput(BaseModelOutput):
+    hidden_states_original_spatial_dimensions: Tuple[Tuple[int, int]] = None
```

Comment:
I don't understand why those new outputs are necessary and what this new field represent. As said before, please redefine the full class (and not rely on inheritance) and include a full docstring.

Reply:
Okay I thought we could inherit from the internal ones (like we did with `ModelOutput`). I need to know the spatial dimensions of each feature map to resize them later on (in the Backbone)